### PR TITLE
Add reindex_no_slots make target (#2947)

### PR DESCRIPTION
**Pull request by noah-aviel-dove**
Author

- [x] PR title references issue
- [x] Title of main commit references issue
- [x] PR is linked to Zenhub issue

Author (reindex)

- [x] Added `r` tag to commit title                         <sub>or this PR does not require reindexing</sub>
- [x] Added `reindex` label to PR                           <sub>or this PR does not require reindexing</sub>

Author (freebies & chains)

- [x] Freebies are blocked on this PR                       <sub>or there are no freebies in this PR</sub>
- [x] Freebies are referenced in commit titles              <sub>or there are no freebies in this PR</sub>
- [x] This PR is blocked by previous PR in the chain        <sub>or this PR is not chained to another PR</sub>
- [x] Added `chain` label to the blocking PR                <sub>or this PR is not chained to another PR</sub>

Author (upgrading)

- [x] Documented upgrading of deployments in UPGRADING.rst  <sub>or this PR does not require upgrading</sub>
- [x] Added `u` tag to commit title                         <sub>or this PR does not require upgrading</sub>
- [x] Added `upgrade` label to PR                           <sub>or this PR does not require upgrading</sub>

Author (requirements, before every review)

- [x] Ran `make requirements_update`                        <sub>or this PR leaves requirements*.txt, common.mk and Makefile untouched</sub>
- [x] Added `R` tag to commit title                         <sub>or this PR leaves requirements*.txt untouched</sub>
- [x] Added `reqs` label to PR                              <sub>or this PR leaves requirements*.txt untouched</sub>

Author (before every review)

- [x] `make integration_test` passes in personal deployment <sub>or this PR does not touch functionality that could break the IT</sub>
- [x] Rebased branch on `develop`, squashed old fixups

Primary reviewer (after approval)

- [x] Commented on demo expectations                        <sub>or labelled PR as `no demo`</sub>
- [x] Decided if PR can be labeled `no sandbox`
- [x] PR title is appropriate as title of merge commit
- [x] Moved ticket to Approved column
- [x] Assigned PR to an operator

Operator (before pushing merge the commit)

- [ ] Checked `reindex` label and `r` commit title tag
- [ ] Checked that Demo expectations are clear              <sub>or PR is labeled as `no demo`</sub>
- [ ] Rebased and squashed branch
- [ ] Sanity-checked history
- [ ] Pushed PR branch to Github
- [ ] Branch pushed to Gitlab and build passes in `sandbox` <sub>or added `no sandbox` label</sub>
- [ ] Started reindex in `sandbox`                          <sub>or this PR does not require reindexing `sandbox`</sub>
- [ ] Checked for failures in `sandbox`                     <sub>or this PR does not require reindexing `sandbox`</sub>
- [ ] Added PR reference to merge commit title
- [ ] Collected commit title tags in merge commit title
- [ ] Moved linked issue to Merged column
- [ ] Pushed merge commit to Github

Operator (after pushing the merge commit)

- [ ] Moved freebies to Merged column                       <sub>or there are no freebies in this PR</sub> 
- [ ] Shortened the PR chain                                <sub>or this PR is not the base of another PR</sub>
- [ ] Verified that `N reviews` labelling is accurate
- [ ] Pushed merge commit to Gitlab                         <sub>or this changes can be pushed later, together with another PR</sub>
- [ ] Deleted PR branch from Github and Gitlab

Operator (reindex) 

- [ ] Started reindex in `dev`                              <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Checked for failures in `dev`                         <sub>or this PR does not require reindexing `dev`</sub>
- [ ] Started reindex in `prod`                             <sub>or this PR does not require reindexing `prod`</sub>
- [ ] Checked for failures in `prod`                        <sub>or this PR does not require reindexing `prod`</sub>

Operator

- [ ] Unassigned PR

### CODE REVIEW

**Comment by noah-aviel-dove** [comment]
Retracting review, spotted a typo

File: Makefile [hunk 1]
```diff
@@ -110,9 +110,15 @@ delete: check_python check_branch
...
 
+reindex_args = --delete --index --purge --partition-prefix-length=2
+
 .PHONY: reindex
 reindex: check_python check_branch
-	python scripts/reindex.py --delete --index --purge --partition-prefix-length=2
+	python scripts/reindex.py ${reindex_args}
+
+.PHONY: small_reindex
+small_reindex: check_python check_branch
```
Comment (jessebrennan) [inline Makefile:120]: Consider a name that starts with `reindex_` which follows the same pattern used by `make requirements*`.

**Review by jessebrennan — APPROVED** [review]
One minor suggestion, but otherwise, this looks good to me.

File: Makefile (hunk 1 above)
Comment (noah-aviel-dove) [reply Makefile:120]: Good catch, Hannes specifically wanted a target that worked with tab-completion.

**Review by jessebrennan — APPROVED** [review]

**Review by hannes-ucsc — APPROVED** [review]

**Comment by hannes-ucsc** [comment]
For demo, show target in action on personal deployment.

**Comment by hannes-ucsc** [comment]
@amarjandu could you make sure this doesn't break the `reindex` job by manually kicking it off in the sandbox?

**Comment by amarjandu** [comment]
This failed due to not having requirements up to date. 
https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/17757

Need to push a commit with updated requirements.
See https://github.com/DataBiosphere/azul/issues/2899#issuecomment-804508017

**Comment by amarjandu** [comment]
@hannes-ucsc the reindex job still passes 🎉 
https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/jobs/17776

**Pull request merged**